go/vt/vttablet/tabletserver/vstreamer/planbuilder.go

```diff
@@ -111,6 +121,67 @@ func (plan *Plan) fields() []*querypb.Field {
 	return fields
 }
 
+// getOpcode returns the equivalent planbuilder opcode for operators that are supported in Filters
+func getOpcode(comparison *sqlparser.ComparisonExpr) (Opcode, error) {
+	var opcode Opcode
+	switch comparison.Operator {
+	case sqlparser.EqualOp:
+		opcode = Equal
+	case sqlparser.LessThanOp:
+		opcode = LessThan
+	case sqlparser.LessEqualOp:
+		opcode = LessThanEqual
+	case sqlparser.GreaterThanOp:
+		opcode = GreaterThan
+	case sqlparser.GreaterEqualOp:
+		opcode = GreaterThanEqual
+	case sqlparser.NotEqualOp:
+		opcode = NotEqual
+	default:
+		return -1, fmt.Errorf("comparison operator %s not supported", comparison.Operator.ToString())
+	}
+	return opcode, nil
+}
+
+// compare returns true after applying the comparison specified in the Filter to the actual data in the column
+func compare(comparison Opcode, columnValue, filterValue sqltypes.Value) (bool, error) {
+	// NullsafeCompare returns 0 if values match, -1 if columnValue < filterValue, 1 if columnValue > filterValue
+	result, err := evalengine.NullsafeCompare(columnValue, filterValue)
```

Comment:
Don't we need a compare that handles null values semantics here?

Reply:
Add null value semantics